internal/server/iigointernal/judiciary.go

```diff
@@ -109,6 +116,18 @@ func (j *judiciary) appointNextPresident(currentPresident shared.ClientID, allIs
 	return nextPresident
 }
 
+// cycleSanctionCache rolls the sanction cahce one turn forward (effectively dropping any sanctions longer than the depth)
+func (j *judiciary) cycleSanctionCache() {
+	newMap := j.localSanctionCache
+	delete(newMap, sanctionCacheDepth-1)
+	newMapCache := newMap
```

Comment:
this assigns a reference to newMap and thus in the for loop below you're essentially working on the same data.
To copy a map you have to use a for loop. Potentially you could create an empty map and assign to it instead

Reply:
Ah crap this is something that needs an immediate fix, ok I'll patch this along with the others and have fun with rebasing again